Fix to get this working with new webpack beta

Klathmon opened the pull request:
This is a small one-line change to get this library working with one of the newer betas of Webpack 2.

They removed `chunk.entry` and replaced it with `chunk.hasRuntime`. This change just does a check to see if `hasRuntime` exists in the chunk, and uses it if it does (falling back to using chunk.entry if it doesn't).

let me know if there are any changes you need me to make before merging, i'll be happy to do them.

jantimon commented:
I am not a maintainer but maybe you could add a comment to explain what's happening there.

jscheid commented:
Sorry, I was travelling and only just saw this. Looks good, thanks!

Could you update `package.json` to reflect the compatibility with that webpack version?

I guess you've tried running the tests manually against that webpack version?

Klathmon commented:
Yeah i ran the tests against webpack 2.1.0-beta.20

I can update the package.json to show compatibility with the new betas now.

And no worries about taking a few days on this, it's not blocking anything of mine so don't feel like you need to rush for me!

jscheid commented:
Thanks for making that change. However, tests are failing here for me. This is what I did:

```
git clone https://github.com/Klathmon/webpack-subresource-integrity.git
cd webpack-subresource-integrity/
git checkout fix-for-webpack-beta 
npm install
npm install webpack@2.1.0-beta.20
npm test
```

I think it will work better using this condition:

```
if (('hasRuntime' in chunk) ? chunk.hasRuntime() : chunk.entry) {
```

Making this change makes it work better, but still not all tests are passing. Could you have a look at fixing it?

jantimon commented:
@jscheid for html-webpack-plugin travis is testing webpack 1 and webpack2:
https://github.com/ampedandwired/html-webpack-plugin/blob/master/.travis.yml

jscheid commented:
Nice @jantimon, thanks! @Klathmon: could you add this to the PR?

Klathmon commented:
I added the .travis.yml changes that @jantimon suggested and I added you (@jscheid) as a collaborator to my branch just in case you want to do anything on your own.

I should have some time later today to take another look at what is going on here.

Klathmon commented:
Well i had a bit of time now after all!

So the first issue was that my hasRuntime check would just blow up if it came back false. Yours works correctly, but all of the tests still fail by never completing.

As it currently stands, it's working correctly in both of my work projects, so that means it's most likely either the testing code that has an issue with webpack 2, or it's something to do with non-entry chunks (since both of my projects don't do any code-splitting).

I'll have time this weekend to really dive into this.

jscheid commented:
I have it working and will push to your branch soon.

jscheid commented:
Seems good now, wdyt?

Klathmon commented:
LGTM 👍  

Thanks for all of that!

jscheid commented:
I'm not quite sure about the peerDependency version spec though. The version spec you use means that 2.1.1 won't be compatible and neither will 2.2. Did you have a reason to include this constraint? I'm in favour of a more optimistic approach where we assume that it will work with all future versions and add exceptions only when it becomes necessary. Thoughts?

jscheid commented:
How about this instead?
```
    "webpack": "^1.12.11 || >=2.1.0-beta"
```

jscheid commented:
Make that `^1.12.11 || >=2.1.0-beta <3.0`

Klathmon commented:
I'm just more pessimistic when it comes to version numbers. I'd rather it announce that it might not work and let the user ignore the warning if they want to.

But that's just a personal preference, and I haven't been using webpack long enough to know how safe they are with their upgrades, so whatever you think is best probably is.

jscheid commented:
But with that line of thinking we should specify `beta.20` since, this being a beta, there is no guarantee it will work with `beta.21`...

I think I'll change it to the optimistic constraints and assume that (a) webpack peeps won't introduce any more breaking changes in the beta and (b) after the 2.1.0 release they'll stick to semver. They've been pretty good at that so far.

FWIW [`html-webpack-plugin` uses `*`](https://github.com/ampedandwired/html-webpack-plugin/blob/9d00ad79ede7bf6d9f631d59338cfdabaea26e86/package.json#L66) so compared to that I'm being conservative ;-)

Klathmon commented:
Yeah betas are a weird area...

My line of thinking is more stuck on the application side where the stricter the better! For a library it makes sense to be more lenient (not quite * lenient though!)

Anyway that is more than fine with me, thanks again for this fix and for the plugin in general!

jscheid commented:
No problem, thanks for your contribution!

Released in version 0.5.0.
